sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_utils.py

```diff
@@ -15,12 +15,59 @@
 from ._generated.models import (
     CommunicationIdentifierModel,
     CommunicationUserIdentifierModel,
-    PhoneNumberIdentifierModel
+    PhoneNumberIdentifierModel,
+    CallLocator
 )
+if TYPE_CHECKING:
+    from ._models import ServerCallLocator, GroupCallLocator
 
-def serialize_identifier(
-        identifier:CommunicationIdentifier
-        ) -> Dict[str, Any]:
+
+def build_call_locator(
+        args: List[Union['ServerCallLocator', 'GroupCallLocator']],
+        kwargs: Dict[str, Any]
+    ) -> CallLocator:
+    """Build the generated callLocator object from args in kwargs with support for legacy models.
+
+    :param args: Any positional parameters provided. This may include the legacy model. The new method signature
+     does not support positional params, so if there's anything here, it's the old model.
+    :type args: list[ServerCallLocator or GroupCallLocator]
+    :param kwargs: Any keyword-only parameters provided. This could include the legacy model (call_locator) or the
+     new flattened args (group_call_id, server_call_id). Only one value can be provided.
+    :type kwargs: dict[str, any]
+    """
+    call_locator: Optional[CallLocator] = None
+    if args:
+        if len(args) > 1:
+            raise TypeError(f"Unexpected positional arguments: {args[1:]}")
+        call_locator = args[0]._to_generated()  # pylint:disable=protected-access
+
+    if "call_locator" in kwargs:
+        if call_locator is not None:
+            raise ValueError(
+                "Received multiple values for call_locator. "
+                "Please provide either 'group_call_id' or 'server_call_id'."
+            )
+        call_locator = kwargs.pop("call_locator")._to_generated()  # pylint:disable=protected-access
+    if "group_call_id" in kwargs:
+        if call_locator is not None:
+            raise ValueError(
+                "Received multiple values for call locator. "
+                "Please provide either 'group_call_id' or 'server_call_id'."
+            )
+        call_locator = CallLocator(group_call_id=kwargs.pop("group_call_id"), kind="groupCallLocator")
+    if "server_call_id" in kwargs:
+        if call_locator is not None:
+            raise ValueError(
+                "Received multiple values for call locator. "
+                "Please provide either 'group_call_id' or 'server_call_id'."
+            )
+        call_locator = CallLocator(server_call_id=kwargs.pop("server_call_id"), kind="serverCallLocator")
+    if call_locator is None:
+        raise ValueError("Call locator required. Please provide either 'group_call_id' or 'server_call_id'.")
+    return call_locator
```

Comment:
should we check for the scenario of both 'group_call_id' and 'server_call_id' provided via kwargs, or is that too unlikely a scenario?

Reply:
This code actually already checks for that scenario - as they're discrete if-statements, not if-else statements. I already added a test for it in `test_call_recording_client.py` below.

Reply:
You're right, I misread the code. Sorry about that!